### ISSUE TITLE
CHE deployment failed: KEYCLOAK pod crashloops in minikube cluster because of POSTGRES(password authentication failure)

### ISSUE THREAD

**Issue by debimishra89** [eclipse-che/che#20394]
### Describe the bug

I am new to kubernetes and trying to install eclipse che on minikube. It requires postgres and keycloak pods in CHECTL DEPLOYMENT stage. but sadly keycloak pod crashloops because of

#### What I tried
I explicitly created password for keycloak user on postgres DB, but no luck.

OS: centos7 VM

cluster: minikube single node cluster (v1.22.0)

postgres pod and database are up and running

keycloak image: quay.io/eclipse/che-keycloak:7.35.1

### Che version

7.35@latest

### Steps to reproduce
```
mkdir kube
cd kube
curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
curl -LO "https://dl.k8s.io/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl.sha256"
curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
sudo install minikube-linux-amd64 /usr/local/bin/minikube
chmod u+x kubectl
minikube start --addons=ingress --vm=true --memory=8192 --driver=none
minikube kubectl – get po -A
chectl server:deploy --platform minikube
```
### Expected behavior

"chectl server:deploy --platform minikube" command should execute successfully

### Runtime

minikube

### Screenshots

![image](https://user-images.githubusercontent.com/84674037/131692803-d83da0b4-aaad-4fde-9939-4d015c5cf141.png)

![image](https://user-images.githubusercontent.com/84674037/131692950-5d640525-7e95-4330-a794-44537a2f6fa9.png)

![image](https://user-images.githubusercontent.com/84674037/131693072-dc9a438b-16a1-4e63-ac73-352c0ae8326a.png)

### Installation method

chectl/latest

### Environment

Linux

### Eclipse Che Logs
```
Eclipse Che logs will be available in '/tmp/chectl-logs/1630508596705'
› Current Kubernetes context: 'minikube'
  ✔ Verify Kubernetes API...OK
  ✔ Verify if namespace 'eclipse-che' exists
  ✔ Read Operator logs...done
  ✔ Read Eclipse Che Server logs...done
  ✔ Read PostgreSQL logs...done
  ✔ Read Keycloak logs...done
  ✔ Read Plug-in Registry logs...done
  ✔ Read Devfile Registry logs...done
  ✔ Read Eclipse Che Dashboard logs...done
  ✔ Read namespace events...done
Command server:logs has completed successfully in 00:00.
```
#### kubectl get events -n eclipse-che
```
LAST SEEN   TYPE      REASON                  OBJECT                                MESSAGE
5m29s       Normal    Scheduled               pod/che-operator-778d679b49-tl4hv     Successfully assigned eclipse-che/che-operator-778d679b49-tl4hv to exp-3.dastc.stee.com
72s         Normal    Pulling                 pod/che-operator-778d679b49-tl4hv     Pulling image "quay.io/eclipse/che-operator:7.35.1"
5m16s       Normal    Pulled                  pod/che-operator-778d679b49-tl4hv     Successfully pulled image "quay.io/eclipse/che-operator:7.35.1" in 5.127437427s
60s         Normal    Created                 pod/che-operator-778d679b49-tl4hv     Created container che-operator
47s         Normal    Started                 pod/che-operator-778d679b49-tl4hv     Started container che-operator
79s         Warning   Unhealthy               pod/che-operator-778d679b49-tl4hv     Readiness probe failed: Get "http://172.17.0.3:6789/readyz": dial tcp 172.17.0.3:6789: connect: connection refused
79s         Warning   Unhealthy               pod/che-operator-778d679b49-tl4hv     Liveness probe failed: Get "http://172.17.0.3:6789/healthz": dial tcp 172.17.0.3:6789: connect: connection refused
65s         Normal    Pulled                  pod/che-operator-778d679b49-tl4hv     Successfully pulled image "quay.io/eclipse/che-operator:7.35.1" in 6.251886764s
5m29s       Normal    SuccessfulCreate        replicaset/che-operator-778d679b49    Created pod: che-operator-778d679b49-tl4hv
5m29s       Normal    ScalingReplicaSet       deployment/che-operator               Scaled up replica set che-operator-778d679b49 to 1
4m56s       Normal    Scheduled               pod/che-tls-job-v46qc                 Successfully assigned eclipse-che/che-tls-job-v46qc to exp-3.dastc.stee.com
4m54s       Normal    Pulled                  pod/che-tls-job-v46qc                 Container image "quay.io/eclipse/che-tls-secret-creator:alpine-d1ed4ad" already present on machine
4m53s       Normal    Created                 pod/che-tls-job-v46qc                 Created container che-tls-job-job-container
4m52s       Normal    Started                 pod/che-tls-job-v46qc                 Started container che-tls-job-job-container
4m56s       Normal    SuccessfulCreate        job/che-tls-job                       Created pod: che-tls-job-v46qc
4m46s       Normal    Completed               job/che-tls-job                       Job completed
3m5s        Normal    Sync                    ingress/che                           Scheduled for sync
5m12s       Normal    LeaderElection          configmap/e79b08a4.org.eclipse.che    che-operator-778d679b49-tl4hv_085a3f8f-125f-4db5-9a4d-6f47342eed61 became leader
31s         Normal    LeaderElection          configmap/e79b08a4.org.eclipse.che    che-operator-778d679b49-tl4hv_012725e0-a2c6-4a6a-9e79-22bd422e0ea6 became leader
3m57s       Normal    Scheduled               pod/keycloak-896c944b6-th2wt          Successfully assigned eclipse-che/keycloak-896c944b6-th2wt to exp-3.dastc.stee.com
9s          Normal    Pulled                  pod/keycloak-896c944b6-th2wt          Container image "quay.io/eclipse/che-keycloak:7.35.1" already present on machine
104s        Normal    Created                 pod/keycloak-896c944b6-th2wt          Created container keycloak
103s        Normal    Started                 pod/keycloak-896c944b6-th2wt          Started container keycloak
23s         Warning   BackOff                 pod/keycloak-896c944b6-th2wt          Back-off restarting failed container
117s        Warning   Unhealthy               pod/keycloak-896c944b6-th2wt          Readiness probe failed: Get "http://172.17.0.6:8080/auth/js/keycloak.js": dial tcp 172.17.0.6:8080: connect: connection refused
65s         Warning   Unhealthy               pod/keycloak-896c944b6-th2wt          Readiness probe failed: HTTP probe failed with statuscode: 503
3m57s       Normal    SuccessfulCreate        replicaset/keycloak-896c944b6         Created pod: keycloak-896c944b6-th2wt
3m5s        Normal    Sync                    ingress/keycloak                      Scheduled for sync
3m57s       Normal    ScalingReplicaSet       deployment/keycloak                   Scaled up replica set keycloak-896c944b6 to 1
4m37s       Warning   FailedScheduling        pod/postgres-69964766d5-scjnz         0/1 nodes are available: 1 pod has unbound immediate PersistentVolumeClaims.
4m35s       Normal    Scheduled               pod/postgres-69964766d5-scjnz         Successfully assigned eclipse-che/postgres-69964766d5-scjnz to exp-3.dastc.stee.com
4m31s       Normal    Pulled                  pod/postgres-69964766d5-scjnz         Container image "quay.io/eclipse/che--centos--postgresql-96-centos7:9.6-b681d78125361519180a6ac05242c296f8906c11eab7e207b5ca9a89b6344392" already present on machine
4m30s       Normal    Created                 pod/postgres-69964766d5-scjnz         Created container postgres
4m28s       Normal    Started                 pod/postgres-69964766d5-scjnz         Started container postgres
80s         Warning   Unhealthy               pod/postgres-69964766d5-scjnz         Readiness probe failed:
4m37s       Normal    SuccessfulCreate        replicaset/postgres-69964766d5        Created pod: postgres-69964766d5-scjnz
4m37s       Normal    Provisioning            persistentvolumeclaim/postgres-data   External provisioner is provisioning volume for claim "eclipse-che/postgres-data"
4m37s       Normal    ExternalProvisioning    persistentvolumeclaim/postgres-data   waiting for a volume to be created, either by external provisioner "k8s.io/minikube-hostpath" or manually created by system administrator
4m37s       Normal    ProvisioningSucceeded   persistentvolumeclaim/postgres-data   Successfully provisioned volume pvc-5af1bfae-3906-4714-a1dc-dd305adc0d25
4m37s       Normal    ScalingReplicaSet       deployment/postgres                   Scaled up replica set postgres-69964766d5 to 1
```

### Additional context
```
error log from keycloak pod:
05:47:24,580 WARN [org.jboss.jca.core.connectionmanager.pool.strategy.OnePool] (ServerService Thread Pool -- 61) IJ000604: Throwable while attempting to get a new connection: null: javax.resource.ResourceException: IJ031084: Unable to create connection
at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createLocalManagedConnection(LocalManagedConnectionFactory.java:345)
at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.getLocalManagedConnection(LocalManagedConnectionFactory.java:352)
at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createManagedConnection(LocalManagedConnectionFactory.java:287)
at org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreConcurrentLinkedDequeManagedConnectionPool.createConnectionEventListener(SemaphoreConcurrentLinkedDequeManagedConnectionPool.java:1325)
at org.jboss.jca.core.connectionmanager.pool.mcp.SemaphoreConcurrentLinkedDequeManagedConnectionPool.getConnection(SemaphoreConcurrentLinkedDequeManagedConnectionPool.java:499)
at org.jboss.jca.core.connectionmanager.pool.AbstractPool.getSimpleConnection(AbstractPool.java:632)
at org.jboss.jca.core.connectionmanager.pool.AbstractPool.getConnection(AbstractPool.java:604)
at org.jboss.jca.core.connectionmanager.AbstractConnectionManager.getManagedConnection(AbstractConnectionManager.java:624)
at org.jboss.jca.core.connectionmanager.tx.TxConnectionManagerImpl.getManagedConnection(TxConnectionManagerImpl.java:440)
at org.jboss.jca.core.connectionmanager.AbstractConnectionManager.allocateConnection(AbstractConnectionManager.java:789)
at org.jboss.jca.adapters.jdbc.WrapperDataSource.getConnection(WrapperDataSource.java:138)
at org.jboss.as.connector.subsystems.datasources.WildFlyDataSource.getConnection(WildFlyDataSource.java:64)
at org.keycloak.connections.jpa.DefaultJpaConnectionProviderFactory.getConnection(DefaultJpaConnectionProviderFactory.java:376)
at org.keycloak.connections.jpa.updater.liquibase.lock.LiquibaseDBLockProvider.lazyInit(LiquibaseDBLockProvider.java:65)
at org.keycloak.connections.jpa.updater.liquibase.lock.LiquibaseDBLockProvider.lambda$waitForLock$0(LiquibaseDBLockProvider.java:97)
at org.keycloak.models.utils.KeycloakModelUtils.suspendJtaTransaction(KeycloakModelUtils.java:678)
at org.keycloak.connections.jpa.updater.liquibase.lock.LiquibaseDBLockProvider.waitForLock(LiquibaseDBLockProvider.java:95)
at org.keycloak.services.resources.KeycloakApplication$1.run(KeycloakApplication.java:144)
at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:227)
at org.keycloak.services.resources.KeycloakApplication.(KeycloakApplication.java:137)
at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
at org.jboss.resteasy.core.ConstructorInjectorImpl.construct(ConstructorInjectorImpl.java:152)
at org.jboss.resteasy.spi.ResteasyProviderFactory.createProviderInstance(ResteasyProviderFactory.java:2750)
at org.jboss.resteasy.spi.ResteasyDeployment.createApplication(ResteasyDeployment.java:364)
at org.jboss.resteasy.spi.ResteasyDeployment.startInternal(ResteasyDeployment.java:277)
at org.jboss.resteasy.spi.ResteasyDeployment.start(ResteasyDeployment.java:89)
at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.init(ServletContainerDispatcher.java:119)
at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.init(HttpServletDispatcher.java:36)
at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:303)
at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:143)
at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:583)
at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:554)
at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:596)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:97)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:78)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
at java.lang.Thread.run(Thread.java:748)
at org.jboss.threads.JBossThread.run(JBossThread.java:485)
Caused by: org.postgresql.util.PSQLException: FATAL: password authentication failed for user "keycloak"
at org.postgresql.core.v3.ConnectionFactoryImpl.doAuthentication(ConnectionFactoryImpl.java:514)
at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:141)
at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:192)
at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
at org.postgresql.jdbc.PgConnection.(PgConnection.java:195)
at org.postgresql.Driver.makeConnection(Driver.java:454)
at org.postgresql.Driver.connect(Driver.java:256)
at org.jboss.jca.adapters.jdbc.local.LocalManagedConnectionFactory.createLocalManagedConnection(LocalManagedConnectionFactory.java:321)
... 55 more

05:47:24,587 INFO [org.jboss.as.server] (Thread-2) WFLYSRV0220: Server shutdown has been requested via an OS signal
05:47:24,593 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 61) MSC000001: Failed to start service jboss.deployment.unit."keycloak-server.war".undertow-deployment: org.jboss.msc.service.StartException in service jboss.deployment.unit."keycloak-server.war".undertow-deployment: java.lang.RuntimeException: RESTEASY003325: Failed to construct public org.keycloak.services.resources.KeycloakApplication(javax.servlet.ServletContext,org.jboss.resteasy.core.Dispatcher)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
at java.lang.Thread.run(Thread.java:748)
at org.jboss.threads.JBossThread.run(JBossThread.java:485)
Caused by: java.lang.RuntimeException: RESTEASY003325: Failed to construct public org.keycloak.services.resources.KeycloakApplication(javax.servlet.ServletContext,org.jboss.resteasy.core.Dispatcher)
at org.jboss.resteasy.core.ConstructorInjectorImpl.construct(ConstructorInjectorImpl.java:164)
at org.jboss.resteasy.spi.ResteasyProviderFactory.createProviderInstance(ResteasyProviderFactory.java:2750)
at org.jboss.resteasy.spi.ResteasyDeployment.createApplication(ResteasyDeployment.java:364)
at org.jboss.resteasy.spi.ResteasyDeployment.startInternal(ResteasyDeployment.java:277)
at org.jboss.resteasy.spi.ResteasyDeployment.start(ResteasyDeployment.java:89)
at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.init(ServletContainerDispatcher.java:119)
at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.init(HttpServletDispatcher.java:36)
at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:303)
at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:143)
at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:583)
at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:554)
at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1502)
at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:596)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:97)
at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:78)
... 8 more
Caused by: java.lang.RuntimeException: Failed to connect to database
at org.keycloak.connections.jpa.DefaultJpaConnectionProviderFactory.getConnection(DefaultJpaConnectionProviderFactory.java:382)
at org.keycloak.connections.jpa.updater.liquibase.lock.LiquibaseDBLockProvider.lazyInit(LiquibaseDBLockProvider.java:65)
at org.keycloak.connections.jpa.updater.liquibase.lock.LiquibaseDBLockProvider.lambda$waitForLock$0(LiquibaseDBLockProvider.java:97)
at org.keycloak.models.utils.KeycloakModelUtils.suspendJtaTransaction(KeycloakModelUtils.java:678)
at org.keycloak.connections.jpa.updater.liquibase.lock.LiquibaseDBLockProvider.waitForLock(LiquibaseDBLockProvider.java:95)
at org.keycloak.services.resources.KeycloakApplication$1.run(KeycloakApplication.java:144)
at org.keycloak.models.utils.KeycloakModelUtils.runJobInTransaction(KeycloakModelUtils.java:227)
at org.keycloak.services.resources.KeycloakApplication.(KeycloakApplication.java:137)
at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
at org.jboss.resteasy.core.ConstructorInjectorImpl.construct(ConstructorInjectorImpl.java:152)
... 31 more
Caused by: java.sql.SQLException: javax.resource.ResourceException: IJ000453: Unable to get managed connection for java:jboss/datasources/KeycloakDS
at org.jboss.jca.adapters.jdbc.WrapperDataSource.getConnection(WrapperDataSource.java:146)
at org.jboss.as.connector.subsystems.datasources.WildFlyDataSource.getConnection(WildFlyDataSource.java:64)
at org.keycloak.connections.jpa.DefaultJpaConnectionProviderFactory.getConnection(DefaultJpaConnectionProviderFactory.java:376)
... 43 more
```

**Comment by debimishra89** [comment]
Hi Team,  Can you please help me with some workaround solution for the time being

**Comment by tolusha** [comment]
@debimishra89 
Try to redeploy Eclipse Che
1. `chectl server:delete -y`
2. `chectl server:deploy --platform minikube`

**Comment by debimishra89** [comment]
> @debimishra89
> Try to redeploy Eclipse Che
> 
> 1. `chectl server:delete -y`
> 2. `chectl server:deploy --platform minikube`

I tried both of these steps. no luck. This error looks like a bug. it doesn't seem to be my VM issue.

**Comment by debimishra89** [comment]
@tolusha , May I know in what situation the below error comes up
#### Caused by: org.postgresql.util.PSQLException: FATAL: password authentication failed for user "keycloak"

**Comment by tolusha** [comment]
Does the output of commands are identical?

1 .

```
kubectl get secret che-identity-secret -n eclipse-che -o json | jq -r '.data.password' | base64 -d
```

2. 

```bash
KEYCKOAK_POD=$(kubectl get pods -n eclipse-che | grep keycloak | awk '{print $1}')
kubectl exec -it $KEYCKOAK_POD -n eclipse-che  -- bash  -c "printenv KEYCLOAK_PASSWORD"
```

**Comment by debimishra89** [comment]
> Does the output of commands are identical?
> 
> 1 .
> 
> ```
> kubectl get secret che-identity-secret -n eclipse-che -o json | jq -r '.data.password' | base64 -d
> ```
> 
> 
> ```shell
> KEYCKOAK_POD=$(kubectl get pods -n eclipse-che | grep keycloak | awk '{print $1}')
> kubectl exec -it $KEYCKOAK_POD -n eclipse-che  -- bash  -c "printenv KEYCLOAK_PASSWORD"
> ```



> ```shell
> kubectl exec -it $KEYCKOAK_POD -n eclipse-che  -- bash  -c "printenv KEYCLOAK_PASSWORD"
> ```

![image](https://user-images.githubusercontent.com/84674037/131848689-0aa1375a-e6b6-4537-8f3c-ba84ade80aca.png)

2nd command is throwing error as keycloak pod id crashlooping

**Comment by debimishra89** [comment]
@tolusha , I redeployed and managed to get the outputs of the above commnads before pod crashes. both command give SAME output.
![image](https://user-images.githubusercontent.com/84674037/131851027-af47c398-d9f8-400a-bb79-bae5d87f3af4.png)

**Comment by tolusha** [comment]
let's update `keycloak` user with a correct password

```bash
PASSWORD=$(kubectl get secret che-identity-postgres-secret -n eclipse-che -o json | jq -r '.data.password' | base64 -d)
POSTGRES_POD=$(kubectl get pods -n eclipse-che | grep postgres | awk '{print $1}')
kubectl exec -it $POSTGRES_POD -n eclipse-che  -- bash  -c "psql postgres -tAc \"ALTER USER keycloak WITH PASSWORD '"$PASSWORD"'\""
```

**Comment by debimishra89** [comment]
> ```shell
> kubectl exec -it $POSTGRES_POD -n eclipse-che  -- bash  -c "psql postgres -tAc \"ALTER USER keycloak WITH PASSWORD '"$PASSWORD"'\""
> ```

it worked. that step passed. thanks much.

**Comment by tolusha** [comment]
Let's keep this issue open.
Because I don't know what the cause is.

**Comment by debimishra89** [comment]
I must say che deployment has lots of intermittent issues. its very unstable on centos VM + Minikube.
I am still not able to deploy successfully because of intermittent failures during
scheduling, downloading and starting steps

**Comment by debimishra89** [comment]
> Let's keep this issue open.
> Because I don't know what the cause is.

successfully deployed CHE. thanks for your help. please take care of this password bug.

**Comment by tolusha** [comment]
I mark this issue as resolved, since 7.42 there is no more Keycloak
